network-api/networkapi/utility/templatetags/primary_active_nav.py

```diff
@@ -4,13 +4,12 @@
 
 
 @register.simple_tag(name='primary_active_nav')
-def primary_active_nav(request, target_path):
+def primary_active_nav(request, root_path, target_path):
 
     if not request:
         return ""
 
-    # remove langauge code segment from path to make path comparison easier
-    lang = f"/{request.LANGUAGE_CODE}"
-    path_info = request.path_info[len(lang):] if request.path_info.startswith(lang) else request.path_info
+    if target_path == root_path:
+        return "active" if request.path == target_path else ""
 
-    return "active" if path_info.rstrip("/") == target_path.rstrip("/") else ""
+    return "active" if request.path.startswith(target_path) else ""
```

Comment:
Following up on above, how does this function know to grab not just the locale?

Reply:
(I mean, this works, so R+, but I'm a bit confused as to why)

Reply:
"if the request path that django is serving, including all the everythings, starts with the same path (including all the everythings) that is associated with the target_path, then this page is active in the menu, because the user's either on this exact page, or on one of its children"

Reply:
Ah. We're comparing strings here. That makes much more sense.